app/src/main/java/com/example/pokemon/presenters/MainPresenter.kt

```diff
@@ -47,29 +53,33 @@ class MainPresenter
     override fun setFavorite(pokemon: Pokemon, buttonState: Boolean) =
         pokemonRepository.setFavoritePokemon(pokemon, buttonState)
 
-    private fun getPokemonsFromRepo(offset: Int) {
-
-        mCompositeDisposable.add(pokemonRepository.getPokemonList(offset)
-            .retry(5)
-            .subscribeOn(Schedulers.io())
-            .observeOn(AndroidSchedulers.mainThread())
-            .subscribe({
-                it?.let {
-                    populateList(it)
-                }
-            }, {
-                mView?.showErrorToast("Error calling API")
-            }
-            ))
+    private fun getPokemonsFromRepo(offset: Int, refresh: Boolean) {
+
+        mCompositeDisposable.add(
+            pokemonRepository.getPokemonList(offset)
+                .retry(5)
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe({
+                    it?.let {
+                        if (refresh) rePopulateList(it)
+                        else populateList(it)
+                    }
+                }, {
+                    mView?.showErrorToast("Error calling API")
+                })
+        )
+        mView?.setRefreshFalse()
```

Comment:
we don't need to add this. Whenever you are adding item to the adapter in activity you can set it to false. Presenter do not need to control it